themes/website/static/js/osdetect.js

```diff
@@ -183,28 +197,60 @@ if (window.navigator.userAgent.indexOf("Windows Phone") != -1) {
 	All the magic stuff the makes the buttons change and stuff
 */
 
+// Set platform header
+document.getElementById("downloadHeader").innerHTML = "Download Tox for " + OSName;
+
 // Set platform toxlock
 document.getElementById("platImg").src = "theme/img/plat/" + OSName.toLowerCase() + ".svg";
 
 // Remove normal button
 document.getElementById("defaultButton").innerHTML = "";
 
+var buttonArea = document.getElementById("buttonArea");
+
+
 // Loop through all links and make buttons and info modals
-for (var i = 0; i < clients.length; i++) {
+for (var i = 0; i < clients.length; i++) 
+{
 	var client = clients[i];
 
-	var button = "\
-	<a id='link" + i + "' href='" + client.dlLink + "' class='button large-button download'>\
-		<span class='fa fa-" + client.icon + "'>&nbsp;</span>" + client.title + "\
-	</a>";
+	if (!client.icon)
+		client.icon = "";
+
+	if (client.faicon)
+		client.faicon = "fa fa-" + client.faicon;
+	else
+		client.faicon = "";
```

Comment:
Wouldn't the following be more concise?

```js
var icon = client.faicon ? 'fa fa-' + client.faicon : '';
client.faicon = icon;
```

Reply:
It would, but won't it be less easy to read?